Increase the maximum number of vCPUs to 64

rneugeba opened the pull request:
Increase the MADT size accordingly. Add a compile time
check to make sure the MADT doesn't overflow into the FADT.

resolves #96

Signed-off-by: Rolf Neugebauer <rolf.neugebauer@docker.com>

ijc25 commented:
I think the ACPI code should error out if the number of vcpus is enough to overflow the MADT. Peter [suggested he would do the same in bhyve](https://docs.freebsd.org/cgi/getmsg.cgi?fetch=130863+0+/usr/local/www/mailindex/archive/2016/freebsd-virtualization/20160918.freebsd-virtualization) so you might be able to grab a hunk of patch from upstream?

AIUI the current generation of Macs top out at 24 logical cores (2x6 physical core x2 hyperthreads each). It not implausible that the next h/w refresh will bump things past 32 (although given Skylake looks to top out at 8 threads it seems like going to exactly 32 is most plausible) maybe we should just bite the bullet and go to max=64 (with corresponding increase to MADT) while we are at it?

rneugeba commented:
Updated the patch to go to 64 vCPUs. Increased the MADT, updated comments and added a compile time check for the MADT size.

Also added the PR text/title to reflect the changes

ijc25 commented:
LGTM
